File: AndroidNDK/app/src/main/cpp/Manager.cpp
//
//  Manager.cpp
//  ProgrammingLanguages
//
//  Created by Martin Kuvandzhiev on 9/11/17.
//  Copyright © 2017 Rapid Development Crew. All rights reserved.
//

#include "Manager.hpp"
#include "to_string.cpp"

Manager::Manager(std::string name, std::string id, int age, float pricePerHour, float productivityCoef, int workExperience, std::vector<Worker*>workers) : Worker(name, id, age, pricePerHour, productivityCoef, workExperience) {
    this->workers = workers;
}

std::map<std::string, std::string> Manager::dictionaryData() {
    std::map<std::string, std::string> dataDict;
    dataDict = Worker::dictionaryData();
    
    for (int i = 0; i < this->workers.size(); i++) {
        dataDict["Worker" + to_string(i)] = this->workers[i]->name;
    }
    return dataDict;
}

File: AndroidNDK/app/src/main/java/com/rapid/androidndktest/data/Manager.java
package com.rapid.androidndktest.data;

import java.util.List;

/**
 * Created by iliyan on 16.09.17.
 */

public class Manager extends Worker {
    private List<Worker> mWorkers;

    public Manager(String name, String personalId, int age, float pricePerHour, float productivityCoef, int workExperience, List<Worker> workers) {
        super(name, personalId, age, pricePerHour, productivityCoef, workExperience);
        mWorkers = workers;
    }

    public List<Worker> getWorkers() {
        return mWorkers;
    }
}
